### PR TITLE
Reduce number of docker services. Improve FPM config

### PR DESCRIPTION
**Pull request by software-opal**
As the name suggests, prune a bunch of unneeded cruft.

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -38,10 +38,10 @@ services:
...
       - mysql:/var/lib/mysql
 
-  redis:
-    image: redis:latest
-    volumes:
-      - redis:/var/lib/redis
+#   redis:
+#     image: redis:latest
+#     volumes:
+#       - redis:/var/lib/redis
```
Comment (liamoneill) [inline docker/docker-compose.yml:44]: Any reason to keep these commented out? Happy if you delete them to remove cruft from the file.

File: docker/configs/php/php-fpm.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+; served. Equivalent to the ApacheMaxClients directive with mpm_prefork.
+; Equivalent to the PHP_FCGI_CHILDREN environment variable in the original PHP
+; CGI. The below defaults are based on a server without much resources. Don't
+; forget to tweak pm.* to fit your needs.
+; Note: Used when pm is set to 'static', 'dynamic' or 'ondemand'
+; Note: This value is mandatory.
+pm.max_children = 20
+pm.min_spare_servers = 4
+pm.max_spare_servers = 10
+pm.max_requests = 100
```
Comment (liamoneill) [inline docker/configs/php/php-fpm.conf:44]: Is this configuration in order to save memory?

**Review by liamoneill — APPROVED** [review]

File: docker/docker-compose.yml (hunk 1 above)
Comment (software-opal) [reply docker/docker-compose.yml]: Not really, I've removed them

File: docker/configs/php/php-fpm.conf (hunk 2 above)
Comment (software-opal) [reply docker/configs/php/php-fpm.conf:44]: PHP FPM was complaining about there not being enough children to respond to all the requests, so I just kinda picked some numbers that sounded right based on my incredibly rusty knowledge of PHP FPM. I was aiming for enough children to serve requests out of the gate with some extra head-room spare, and then frequent enough replacements for any memory leaks to not be an issue, plus it's localdev so  it's not going to impact us that much.

**Pull request merged**